### ISSUE TITLE
Get logged out moments after trying to log in via pod.movim.eu

### ISSUE THREAD

**Issue by esijg** [movim/movim#35]
Whenever I try to log in to pod.movim.eu, I manage to get in and see the usual home page, but shortly after or when I try to click anything (like "contacts") I get logged out and thrown back to the login screen.

At one point I started getting "invalid password format" error after clicking "Come In!" but before seeing the home page. For some reasons that has stopped, but now I see a error msg on the login screen after being logged out saying "Movim cannot talk with the server, please try again later (Websocket connection error)"

I am using Dreamhosts XMPP feature which might be the source of this, or not. Hard to tell. But I can manage to connect easily via Pidgin, Empathy and other XMPP clients.

**Comment by edhelas** [comment]
Thanks for this report ! This issue seems to be linked with this one https://github.com/edhelas/movim/issues/6 I'll try to get more info ;)

**Comment by src386** [comment]
Same here.
But I don't use self signed certificates.
It's a certificate from StartSSL.

**Comment by edhelas** [comment]
I've debugged a bit your issue. First I have a strange behaviour regarding this line in PHP 

```
dns_get_record('_xmpp-client._tcp.johannesg.com', DNS_SRV)
```

It take more than 10sec to resolve the DNS and return an empty array which is weird because I can reach it instantly using dig

```
dig SRV _xmpp-client._tcp.johannesg.com

; <<>> DiG 9.9.5-11-Debian <<>> SRV _xmpp-client._tcp.johannesg.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 20480
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;_xmpp-client._tcp.johannesg.com. IN    SRV

;; AUTHORITY SECTION:
johannesg.com.      388 IN  SOA ns1.dreamhost.com. hostmaster.dreamhost.com. 2015072700 14431 1800 1814400 14400

;; Query time: 102 msec
;; SERVER: 80.69.67.66#53(80.69.67.66)
;; WHEN: Fri Aug 21 18:19:05 CEST 2015
;; MSG SIZE  rcvd: 121
```

**Comment by src386** [comment]
Is there a way to get an older snapshot of Movim ? My yunohost package is broken.

**Comment by esijg** [comment]
Tried logging in today and it was a bit different this time around. Actually managed to connect and load the contacts page. Stayed online for about 15 seconds maybe (longer than last time) but was kicked out after that with the same error (websocket error).

Sorry for not being able to write more detailed bug reports but my experience with debugging this kind of stuff is pretty much zero.

**Comment by dragondaddy** [comment]
I had a similar problem when I was trying to set up an install of Movim in a LXC container. I could connect but after a few seconds I was sent back to the login page, disconnected.

Turns out it was a websocket issue. The web server in the LXC container that hosts Movim was configured as described in the wiki, but that wasn't enough. I had to add a line for /ws/ in the Nginx server used as a reverse proxy on the main host.

I don't know if it could have anything in common with your problem, but the behavior was quite the same.

**Comment by src386** [comment]
@dragondaddy : what do you mean adding /ws ?
http://wiki.movim.eu/fr:install_new#pour_les_serveurs_nginx 
I'm interested in solving that problem, did you had something in your nginx vhost ?

**Comment by dragondaddy** [comment]
Ok so here's how it works:
- Movim is running in a LXC container, on an Apache server, configured just as recommended in the wiki,
- On the host, I have Nginx used as a reverse proxy.

At first I had no `location /ws/` section in my Nginx configuration. I thought it was OK, because on the login page I had no websocket related error message. But when after logging in, I was disconnected after a few seconds (just as johannesgunnar describes in his 2nd message).

I was able to solve this adding a `location /ws/` section in my Nginx configuration file, just as describe in the wiki, except that instead of writing this
`proxy_pass https://localhost:8080/;`
I had to put this
`proxy_pass https://1.2.3.4;`
(1.2.3.4 being the local address of the LXC container)

Anyway, I don't think it has anything to do with johannesgunnar's problem, except that it is a websocket connection loss.

**Comment by esijg** [comment]
I think I've managed to fix this from my end as the problem was with dreamhost rather than movim (although a more descriptive error msg in movim would help) as my certificate was set on one of my subdomains rather than my main domain. After changing that I managed to log in to movim easily.

Still odd that movim would complain but not other XMPP clients. Or maybe that's a security flaw with other clients and Movim is just too good. ;)
